eva-stats/stats_variant_warehouse.py

```diff
@@ -0,0 +1,181 @@
+        filename = row[4]
+        analyses.append(analysis)
+        analysis_filename[analysis] = filename
+        # All except last row
+        if index < len(result_evapro) - 1:
+            next_row = result_evapro[index + 1]
+            if row[1] != next_row[1]:
+                # if next row is a new project
+                assembly = row[0]
+                project = row[1]
+                get_counts_from_variant_warehouse(assembly, project, analyses, metadata_handle, mongo_handle)
+                get_dates_from_variant_warehouse(assembly, project, analysis_filename, metadata_handle, mongo_handle)
+                analyses = []
+                analysis_filename = {}
+        else:
+            assembly = row[0]
+            project = row[1]
+            get_counts_from_variant_warehouse(assembly, project, analyses, metadata_handle, mongo_handle)
+            get_dates_from_variant_warehouse(assembly, project, analysis_filename, metadata_handle, mongo_handle)
+
+
+def insert_into_stats(metadata_handle, row):
+    assembly, project, analysis, file_id, filename, file_type, taxonomy_id = row[0], row[1], row[2], row[3], row[4], \
+                                                                             row[5], row[6]
+    check_exist_query = "select * from eva_stats.stats where assembly_accession = '{0}' and project_accession = '{1}'" \
+                        " and analysis_accession = '{2}'".format(assembly, project, analysis)
+    insert_query = "insert into eva_stats.stats(assembly_accession, project_accession, analysis_accession, file_id, " \
+                   "filename, file_type, taxonomy_id) values ('{0}','{1}','{2}','{3}','{4}','{5}', {6})".format(
+        assembly, project, analysis, file_id, filename, file_type, taxonomy_id)
+    if len(get_all_results_for_query(metadata_handle, check_exist_query)) == 0:
+        logger.info("Insert data for {0}, {1}, {2} in eva_stats table".format(assembly, project, analysis))
+        execute_query(metadata_handle, insert_query)
+    else:
+        logger.info("Already exists {0}, {1}, {2} in eva_stats table".format(assembly, project, analysis))
+
+
+def get_counts_from_variant_warehouse(assembly, project, analyses, metadata_handle, mongo_handle):
+    database_name = get_database_name_from_assembly(metadata_handle, assembly)
+    logger.info("Database name to query: {0}".format(database_name))
+    logger.info("Getting counts from variants collection for {0}, {1}, {2}".format(assembly, project, analyses))
+    variants_collection = mongo_handle[database_name]['variants_2_0']
+    pipeline = [
+        {"$match": {"files.sid": project, "files.fid": {"$in": analyses}}},
+        {"$project": {"_id": 0, "files.fid": 1}},
+        {"$unwind": "$files"},
+        {"$unwind": "$files.fid"},
+        {"$match": {"files.fid": {"$in": analyses}}},
+        {"$group": {"_id": "$files.fid", "count": {"$sum": 1}}},
+        {"$project": {"_id": 0, "files.fid": "$_id", "count": 1}}
+    ]
```

Comment:
Again, suggest using **allowDiskUse=True** in the pipeline to avoid memory issues...

Reply:
Thanks, added